test/hotspot/jtreg/compiler/intrinsics/float16/TestFP16ScalarAdd.java

```diff
@@ -74,4 +90,45 @@ public void test2() {
         Float16 hf4 = Float16.valueOf((short)17408);
         res = Float16.sum(Float16.sum(Float16.sum(Float16.sum(hf0, hf1), hf2), hf3), hf4).float16ToRawShortBits();
     }
+
+    // Test for optimizing sequence - "dst (ReinterpretS2HF (ConvF2HF src)" in the backend to a single convert
+    // operation and do away with redundant moves
+    @Test
+    @IR(applyIfCPUFeature = {"avx512_fp16" , "true"}, counts = {IRNode.CONVF2HFANDS2HF, " >= 1"})
+    @IR(applyIfCPUFeatureAnd = {"fphp", "true", "asimdhp", "true"}, counts = {IRNode.CONVF2HFANDS2HF, " >= 1"})
+    public void test3() {
```

Comment:
"test3" and "test4" are not really related to FP16 add operation and I can move them to another file if needed.